File: arc134/a/main.py
import math
n, l, w = map(int, input().split())
A = list(map(int, input().split()))

start = 0
ans = 0
for i in range(n):
    if w == 1:
        ans += max(A[i]-start, 0)
    else:
        ans += max(math.ceil((A[i]-start)/w), 0)
    start = A[i]+w

if A[n-1]+w < l:
    if w == 1:
        ans += l-(A[n-1]+w)
    else:
        ans += math.ceil((l-(A[n-1]+w))/w)

print(ans)

File: arc134/a/main.cc
#include <iostream>
#include <vector>
using ll = long long;
using namespace std;
int main(){
    int n;
    cin>>n;
    ll L,w;
    cin>>L>>w;
    ll r = 0;
    vector<ll> a(n+1,L);
    for(int i=0;i<n;i++)cin>>a[i];
    ll ans = 0L;
    for(auto x:a)
    {
        if (x > r)
            ans += (x - r + w- 1) / w;
        r = x + w;
    }
    cout<<ans<<endl;
}